src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -337,9 +362,9 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
         }
     }
 
-    private boolean canReadFrom(Job<?, ?> job, AbstractBuild<?, ?> build) {
-        if ((job instanceof AbstractProject) && CopyArtifactPermissionProperty.canCopyArtifact(
-                build.getProject().getRootProject(),
+    private boolean canReadFrom(Job<?, ?> job, Run<?, ?> build) {
+        if ((job instanceof AbstractProject) && (build instanceof AbstractBuild) && CopyArtifactPermissionProperty.canCopyArtifact(
+                ((AbstractBuild)build).getProject().getRootProject(),
                 ((AbstractProject<?,?>)job).getRootProject()
```

Comment:
Still want to do the `canCopyArtifact` check on non-`AbstractBuild`s, just do not need to call `getRootProject`.